### PR TITLE
[!!!][FEATURE] Use LanguageAspect in QuerySettings for Extbase

### PR DESCRIPTION
**Pull request by linawolf**
resolves https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/132

### CODE REVIEW

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst [hunk 1]
```diff
@@ -152,87 +152,32 @@ Typo3QuerySettings and localization
 
 Extbase renders the translated records in the same way as TypoScript rendering.
 
-..  versionchanged:: 11.0
-    Setting :php:`Typo3QuerySettings->languageMode`  was deprecated and does
-    **not** influence how Extbase queries records.
+..  versionchanged:: 12.0
+    The Extbase query settings rely on
+    :php:`TYPO3\CMS\Core\Context\LanguageAspect` now.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst:157]:
```suggestion
    :php:`\TYPO3\CMS\Core\Context\LanguageAspect` now.
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst [hunk 2]
```diff
@@ -152,87 +152,32 @@ Typo3QuerySettings and localization
...
-..  versionchanged:: 11.0
-    Setting :php:`Typo3QuerySettings->languageMode`  was deprecated and does
-    **not** influence how Extbase queries records.
+..  versionchanged:: 12.0
+    The Extbase query settings rely on
+    :php:`\TYPO3\CMS\Core\Context\LanguageAspect` now.
 
-:php:`Typo3QuerySettings->languageOverlayMode` = true
------------------------------------------------------
+The following methods can be used to set and get the language aspect from any
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst:159]:
```suggestion
The following methods can be used to set and get the :ref:`language aspect <context_api_aspects_language>`  from any
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst [hunk 3]
```diff
@@ -152,87 +152,32 @@ Typo3QuerySettings and localization
 
 Extbase renders the translated records in the same way as TypoScript rendering.
 
-..  versionchanged:: 11.0
-    Setting :php:`Typo3QuerySettings->languageMode`  was deprecated and does
-    **not** influence how Extbase queries records.
+..  versionchanged:: 12.0
+    The Extbase query settings rely on
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst:156]: We should mention here that the old methods still work in v12, but are deprecated.

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
+
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php:2]: We should add strict types header.

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+namespace MyVendor\MyExtension\Domain\Repository;
+
+use TYPO3\CMS\Core\Context\LanguageAspect;
+use TYPO3\CMS\Extbase\Persistence\Repository;
+
+class MyRepository extends Repository
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php:8]:
```suggestion
final class MyRepository extends Repository
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php (hunk 4 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Extbase/Reference/Domain/_Repository/_LanguageAspect.php:2]: done

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**